pkg/api/types.go

```diff
@@ -1309,3 +1317,27 @@ type ResourceQuotaList struct {
 	// Items is a list of ResourceQuota objects
 	Items []ResourceQuota `json:"items"`
 }
+
+// Secret holds secret data of a certain type
+type Secret struct {
+	TypeMeta   `json:",inline"`
+	ObjectMeta `json:"metadata,omitempty"`
+
+	Data map[string][]byte `json:"data,omitempty"`
```

Comment:
http://golang.org/pkg/encoding/json/

    Array and slice values encode as JSON arrays, except that `[]byte` encodes as a base64-encoded string, and a nil slice encodes as the null JSON object.